packages/vite/src/node/ssr/runtime/index.ts

```diff
@@ -0,0 +1,28 @@
+// this file should re-export only things that don't rely on Node.js or other runtime features
+
+export { ModuleCacheMap } from './moduleCache'
+export { ViteRuntime } from './runtime'
+export { ESModulesRunner } from './esmRunner'
+
+export { handleHMRUpdate, createHMRHandler } from './hmrHandler'
```

Comment:
I think we should be marking these APIs with `@experimental` jsdoc.

Also is there a reason `handleHMRUpdate` and `createHMRHandler` is exported here?

Reply:
We can probably remove them now from the exports, in the previous revision it was the framework author's responsibility to call these methods

Reply:
Please don't consider this anything critical, but I happen to rely on `handleHMRUpdate` in my Workerd integration experiment https://github.com/hi-ogawa/vite-plugins/pull/155/files#diff-3928ea75f2cbb10c4b70afd47be15e5aa5d488ce93aebfff69b6d634036ee30e

Probably the expected usage is to provide a proper `HMRRuntimeConnection` when constructing `ViteRuntime`, but right now, I ended up keeping similar structure as my previous vite-node integration, which is to collect `HMRPayload` outside of `HMRRuntimeConnection` and manually call `handleHMRUpdate`.
(previously it was collecting "ssr invalidates" then calling vite-node's `moduleCache.invalidateDepTree`).

I'm not sure if actual framework can get away without having a proper `HMRRuntimeConnection` implemented, but for former adopters of vite-node, exposing `handleHMRUpdate` might make sense.